src/server/game/Pools/QuestPools.cpp

```diff
@@ -0,0 +1,296 @@
+                continue;
+            }
+
+            uint32 questId = fields[0].GetUInt32();
+            uint32 poolId = fields[1].GetUInt32();
+            uint32 poolIndex = fields[2].GetUInt8();
+            uint32 numActive = fields[3].GetUInt32();
+
+            Quest const* quest = sObjectMgr->GetQuestTemplate(questId);
+            if (!quest)
+            {
+                TC_LOG_ERROR("sql.sql", "Table `quest_pool_members` contains reference to non-existing quest %u. Skipped.", questId);
+                continue;
+            }
+            if (!quest->IsDailyOrWeekly() && !quest->IsMonthly())
+            {
+                TC_LOG_ERROR("sql.sql", "Table `quest_pool_members` contains reference to quest %u, which is neither daily, weekly nor monthly. Skipped.", questId);
+                continue;
+            }
+
+            auto& pair = lookup[poolId];
+            if (!pair.first)
+            {
+                pair.first = (quest->IsDaily() ? &_dailyPools : quest->IsWeekly() ? &_weeklyPools : &_monthlyPools);
+                pair.first->emplace_back();
+                pair.second = (pair.first->size()-1);
+
+                pair.first->back().poolId = poolId;
+                pair.first->back().numActive = numActive;
+            }
+
+            auto& members = (*pair.first)[pair.second].members;
+            if (!(poolIndex < members.size()))
+                members.resize(poolIndex+1);
+            members[poolIndex].push_back(questId);
+        } while (result->NextRow());
+    }
+
+    // load saved spawns from character DB
+    {
+        QueryResult result = CharacterDatabase.Query("SELECT pool_id, quest_id FROM pool_quest_save");
+        if (result)
+        {
+            std::unordered_set<uint32> unknownPoolIds;
+            do
+            {
+                Field* fields = result->Fetch();
+
+                uint32 poolId = fields[0].GetUInt32();
+                uint32 questId = fields[1].GetUInt32();
+
+                auto it = lookup.find(poolId);
+                if (it == lookup.end() || !it->second.first)
+                {
+                    TC_LOG_ERROR("sql.sql", "Table `pool_quest_save` contains reference to non-existant quest pool %u. Deleted.", poolId);
+                    unknownPoolIds.insert(poolId);
+                    continue;
+                }
+
+                (*it->second.first)[it->second.second].activeQuests.insert(questId);
+            } while (result->NextRow());
+
+            SQLTransaction trans = CharacterDatabase.BeginTransaction();
+            for (uint32 poolId : unknownPoolIds)
+            {
+                PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_DEL_POOL_QUEST_SAVE);
+                stmt->setUInt32(0, poolId);
+                trans->Append(stmt);
+            }
+            CharacterDatabase.CommitTransaction(trans);
+        }
+    }
+
+    // post-processing and sanity checks
+    for (auto pair : lookup)
```

Comment:
You create container copies here

Reply:
no, pair is `std::pair<uint32, std::pair<std::vector<QuestPool>*, uint32>>`

Reply:
Exactly, you are  creating copies of the pair => copies of the vector
And your type is slightly incorrect (would also result in copies if typed out instead of auto), its `std::pair<uint32 const, std::pair<std::vector<QuestPool>*, uint32>>`

Reply:
pointer to vector